### PR TITLE
add manual job run button

### PR DESCRIPTION
**Pull request by d-g-town**
Screenshot 2023-11-06 at 3 02 25 PM
Screenshot 2023-11-06 at 3 02 38 PM

### CODE REVIEW

File: api/server/handlers/porter_app/app_run.go [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+	ctx, span := telemetry.NewSpan(r.Context(), "serve-app-run")
+	defer span.End()
+
+	project, _ := ctx.Value(types.ProjectScope).(*models.Project)
+	cluster, _ := ctx.Value(types.ClusterScope).(*models.Cluster)
+
+	telemetry.WithAttributes(span,
+		telemetry.AttributeKV{Key: "project-id", Value: project.ID},
+		telemetry.AttributeKV{Key: "cluster-id", Value: cluster.ID},
+	)
```
Comment (ferozemohideen) [inline api/server/handlers/porter_app/app_run.go:63]: is this needed

File: dashboard/src/main/home/app-dashboard/validate-apply/jobs/TriggerJobButton.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+          `/apps/${appName}/job-history?job_run_id=${jobRunID}&service=${jobName}`
+        );
+      })
+      .catch(() => {
+        setStatus("");
+        setErrorMessage("Unable to run job");
+        showIntercomWithMessage({
+          message: "I am running into an issue running my job.",
+        });
+      });
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/jobs/TriggerJobButton.tsx:48]: let's try to default to async/await syntax, wrapping this in a try/catch/finally to properly loading and error states

File: dashboard/src/main/home/app-dashboard/validate-apply/jobs/utils.ts [hunk 3]
```diff
@@ -31,4 +33,22 @@ export const getDuration = (jobRun: JobRun): string => {
...
+        },
+        {
+            project_id: projectId,
+            cluster_id: clusterId,
+            porter_app_name: appName,
+        })
+
+    const parsed = await z.object({job_run_id: z.string()}).parseAsync(resp.data)
+
+    return parsed.job_run_id
```
Comment (ianedwards) [inline dashboard/src/main/home/app-dashboard/validate-apply/jobs/utils.ts:53]: wrap in try catch so the error doesn't propagate up

**Review by ianedwards — APPROVED** [review]
cool stuff

**Review by ferozemohideen — APPROVED** [review]
lgtm, would just change Run job to 'Run once' and would be good to add an icon

File: api/server/handlers/porter_app/app_run.go (hunk 1 above)
Comment (d-g-town) [reply api/server/handlers/porter_app/app_run.go]: no

**Pull request merged**